### DISCUSSION TITLE
[✨] Adding a signal dispatcher for component like in vue

### DISCUSSION THREAD

**Discussion by astraldev** [QwikDev/qwik-evolution#184] [Proposals For Qwik]
### Is your feature request related to a problem?

I want to create a component that emits a signal whenever a button is clicked and the event should be handled by the parent component but I am using **@builder.io/mitosis** to build. Other than that, it is practically impossible for a component to emit a signal. 

### Describe the solution you'd like

Creating a hook for signal emission or just using a `this.emit('<SIGNALNAME>')` to emit a signal

### Describe alternatives you've considered

None

### Additional context

If possible, able to accept mitosis integration

**Comment by wmertens** [comment]
@astraldev can't you pass a signal via context or props, and then the child component can change the value, which notifies every listener including parents?

**Comment by gioboa** [comment]
@astraldev is it solved?

**Comment by astraldev** [comment]
This was a feature request, I am yet to test it. 
If there's a need to close the issue, then fine, although a test would be made in like 2 week

**Comment by gioboa** [comment]
I see, thanks @astraldev 
I move this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
